subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/DefaultXcodeProject.java

```diff
@@ -16,47 +16,31 @@
 
 package org.gradle.ide.xcode.internal;
 
-import org.gradle.api.file.FileCollection;
+import org.gradle.api.file.ConfigurableFileCollection;
 import org.gradle.ide.xcode.XcodeProject;
 import org.gradle.ide.xcode.internal.xcodeproj.GidGenerator;
 
 import java.io.File;
-import java.util.ArrayList;
 import java.util.Collections;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Set;
 
 public class DefaultXcodeProject implements XcodeProject {
     private final GidGenerator gidGenerator = new GidGenerator(Collections.<String>emptySet());
-    private final List<File> additionalFiles = new ArrayList<File>();
     private XcodeTarget target;
-    private FileCollection sources;
+    private ConfigurableFileCollection sources;
```

Comment:
This should be initialised with an empty instance and made final. The `setSources()` method should call the `setFrom()` method on this instance. Otherwise, we're mutating the instance we were supplied as a parameter to `setSources()`, which is not a good idea. And it assumes that `setSources()` is called before `addSourceFile()`.